test/common/sendgrid_credentials_valid.go

```diff
@@ -0,0 +1,38 @@
+package common
+
+import (
+	"crypto/tls"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/smtp"
+	"testing"
+)
+
+func TestSendgridCredentialsAreValid(t *testing.T, ctx *TestingContext) {
+	// Get SMTP secret from rhmi-operator namespace
+	kc := ctx.KubeClient
+	smtpSecret, err := kc.CoreV1().Secrets(RHMIOperatorNamespace).Get("redhat-rhmi-smtp", metav1.GetOptions{})
+	if err != nil {
+		t.Fatal("Failed to get an SMTP secret", err)
+	}
+
+	username, password, host := string(smtpSecret.Data["username"]), string(smtpSecret.Data["password"]), string(smtpSecret.Data["host"])
+
+	if host != "smtp.sendgrid.net" {
+		t.Skip("Skipping the test, because the credentials were not set by a Sendgrid service")
+	}
+
+	// Test if SMTP credentials are valid using smtp.Auth method
+	auth := smtp.PlainAuth("", username, password, host)
+	client, err := smtp.Dial(host + ":587")
```

Comment:
Should we use `smtpSecret.Data["port"]` instead :thinking:

Reply:
thanks, totally missed it